diff --git a/Project/CMake/CMakeLists.txt b/Project/CMake/CMakeLists.txt
index 6c0eb05..1f1def9 100644
--- a/Project/CMake/CMakeLists.txt
+++ b/Project/CMake/CMakeLists.txt
@@ -387,6 +387,70 @@ endif()
 
 # disable some features. Maybe it should be enabled.
 target_compile_definitions(mediainfo PRIVATE MEDIAINFO_LIBMMS_NO PRIVATE MEDIAINFO_GRAPHVIZ_NO)
+# Flags in order as defined in configure.ac.
+target_compile_definitions(mediainfo
+  PRIVATE
+  # Global flags
+  MEDIAINFO_MINIMAL_YES
+  MEDIAINFO_CONFORMANCE_YES # Bug in the minimal configuration.
+  MEDIAINFO_DEMUX_YES # Fix some .wav parsing
+  MEDIAINFO_EVENTS_YES # Required for DEMUX
+  MEDIAINFO_ARCHIVE_NO
+  MEDIAINFO_IMAGE_NO
+  MEDIAINFO_TEXT_NO
+
+  # Disable all tag except the id3
+  MEDIAINFO_APETAG_NO
+  MEDIAINFO_ICC_NO
+  MEDIAINFO_LYRICS3_NO
+  MEDIAINFO_LYRICS3V2_NO
+  MEDIAINFO_PROPERTYLIST_NO
+  MEDIAINFO_SPHERICALVIDEO_NO
+  MEDIAINFO_VORBISCOM_NO
+  MEDIAINFO_XMP_NO
+
+  # Disable some "Multiple" formats
+  MEDIAINFO_ANCILLARY_NO
+  MEDIAINFO_BDAV_NO
+  MEDIAINFO_BDMV_NO
+  MEDIAINFO_CDXA_NO
+  MEDIAINFO_DASHMPD_NO
+  MEDIAINFO_DCP_NO
+  MEDIAINFO_DPG_NO
+  MEDIAINFO_REFERENCES_NO
+  MEDIAINFO_DVDIF_NO
+  MEDIAINFO_DVDV_NO
+  MEDIAINFO_DXW_NO
+  MEDIAINFO_FLV_NO
+  MEDIAINFO_GXF_NO
+  MEDIAINFO_HDSF4M_NO
+  MEDIAINFO_HLS_NO
+  MEDIAINFO_MIXML_NO
+  MEDIAINFO_MXF_NO
+  MEDIAINFO_NUT_NO
+  MEDIAINFO_PMP_NO
+  MEDIAINFO_SKM_NO
+  MEDIAINFO_SWF_NO
+  MEDIAINFO_VBI_NO
+
+  # Disable some "Video" formats
+  MEDIAINFO_AIC_NO
+  MEDIAINFO_AVSV_NO
+  MEDIAINFO_CANOPUS_NO
+  MEDIAINFO_FFV1_NO
+  MEDIAINFO_FLIC_NO
+  MEDIAINFO_HUFFYUV_NO
+  MEDIAINFO_VC3_NO
+  MEDIAINFO_Y4M_NO
+
+  # Disable some "Audio" formats
+  MEDIAINFO_ADIF_NO
+  MEDIAINFO_ADPCM_NO
+  MEDIAINFO_CELT_NO
+  MEDIAINFO_DAT_NO
+  MEDIAINFO_PS2A_NO
+  MEDIAINFO_TTA_NO
+)
 if(NOT CURL_FOUND)
   target_compile_definitions(mediainfo PRIVATE MEDIAINFO_LIBCURL_NO)
 else()
