diff --git a/BUILD.gn b/BUILD.gn
index 6885fc27d..f4533ee76 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -22,6 +22,7 @@ config("pdfium_common_config") {
   defines = [
     "PNG_PREFIX",
     "PNG_USE_READ_MACROS",
+    "FPDFSDK_EXPORTS"
   ]
 
   if (!use_system_libopenjpeg2) {
@@ -132,7 +133,7 @@ jumbo_source_set("pdfium_public_headers") {
   public_configs = [ ":pdfium_public_config" ]
 }
 
-jumbo_static_library("pdfium") {
+shared_library("pdfium") {
   sources = [
     "fpdfsdk/fpdf_annot.cpp",
     "fpdfsdk/fpdf_attachment.cpp",
@@ -202,6 +203,9 @@ jumbo_static_library("pdfium") {
       "gdi32.lib",
       "user32.lib",
     ]
+    sources += [
+      "resources.rc"
+    ]
   }
 
   if (is_mac) {
diff --git a/public/fpdfview.h b/public/fpdfview.h
index 7d78186ac..73a3e1fbb 100644
--- a/public/fpdfview.h
+++ b/public/fpdfview.h
@@ -154,13 +154,22 @@ typedef int FPDF_ANNOT_APPEARANCEMODE;
 // Dictionary value types.
 typedef int FPDF_OBJECT_TYPE;
 
-#if defined(_WIN32) && defined(FPDFSDK_EXPORTS)
-// On Windows system, functions are exported in a DLL
-#define FPDF_EXPORT __declspec(dllexport)
-#define FPDF_CALLCONV __stdcall
+#ifdef FPDFSDK_EXPORTS
+# if defined(_WIN32)
+#  define FPDF_EXPORT __declspec(dllexport)
+#  define FPDF_CALLCONV __stdcall
+# else
+#  define FPDF_EXPORT __attribute__((visibility("default")))
+#  define FPDF_CALLCONV
+# endif
 #else
-#define FPDF_EXPORT
-#define FPDF_CALLCONV
+# if defined(_WIN32)
+#  define FPDF_EXPORT __declspec(dllimport)
+#  define FPDF_CALLCONV __stdcall
+# else
+#  define FPDF_EXPORT
+#  define FPDF_CALLCONV
+# endif
 #endif
 
 // Exported Functions
