diff --git a/portfile.cmake b/portfile.cmake
index 0297760..d31ffeb 100644
--- a/portfile.cmake
+++ b/portfile.cmake
@@ -25,7 +25,7 @@ foreach(feature IN ITEMS "schannel" "sspi" "tool" "winldap")
     endif()
 endforeach()
 
-if("sectransp" IN_LIST FEATURES AND NOT VCPKG_TARGET_IS_OSX)
+if("sectransp" IN_LIST FEATURES AND NOT VCPKG_TARGET_IS_OSX AND NOT VCPKG_TARGET_IS_IOS)
     message(FATAL_ERROR "sectransp is not supported on non-Apple platforms")
 endif()
 
