diff --git a/portfile.cmake b/portfile.cmake
index d0d89fd..a9aff25 100644
--- a/portfile.cmake
+++ b/portfile.cmake
@@ -24,7 +24,7 @@ foreach(feature "schannel" "sspi" "tool")
     endif()
 endforeach()
 
-if("sectransp" IN_LIST FEATURES AND NOT VCPKG_TARGET_IS_OSX)
+if("sectransp" IN_LIST FEATURES AND NOT VCPKG_TARGET_IS_OSX AND NOT VCPKG_TARGET_IS_IOS)
     message(FATAL_ERROR "sectransp is not supported on non-Apple platforms")
 endif()
 
@@ -186,4 +186,4 @@ elseif(VCPKG_TARGET_IS_LINUX)
 endif()
 
 file(INSTALL ${CURRENT_PORT_DIR}/vcpkg-cmake-wrapper.cmake DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
-file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
\ No newline at end of file
+file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
