diff --git a/portfile.cmake b/portfile.cmake
index 9ba7de91b..f1a559d2f 100644
--- a/portfile.cmake
+++ b/portfile.cmake
@@ -360,6 +360,9 @@ if (VCPKG_TARGET_ARCHITECTURE STREQUAL "arm" OR VCPKG_TARGET_ARCHITECTURE STREQU
         endforeach(GAS_PATH)
     endif()
 elseif (VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
+    if(VCPKG_TARGET_IS_OSX)
+        set(OPTIONS_CROSS " --enable-cross-compile --target-os=darwin --arch=x86_64 --extra-ldflags=-arch --extra-ldflags=x86_64 --extra-cflags=-arch --extra-cflags=x86_64 --extra-cxxflags=-arch --extra-cxxflags=x86_64")
+    endif()
 elseif (VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
 else()
     message(FATAL_ERROR "Unsupported architecture")
