diff --git a/portfile.cmake b/portfile.cmake
index 6409d7485..f99f3dd9e 100644
--- a/portfile.cmake
+++ b/portfile.cmake
@@ -53,4 +53,21 @@ vcpkg_cmake_config_fixup()
 
 vcpkg_copy_pdbs()
 
+file(READ "${CURRENT_PACKAGES_DIR}/include/FreeImage.h" _header_data)
+string(REPLACE
+    "typedef int32_t BOOL;"
+    "#define BOOL int32_t"
+    _header_data
+    "${_header_data}"
+)
+
+string(REPLACE
+    "#endif // FREEIMAGE_H"
+    "#undef BOOL
+    #endif // FREEIMAGE_H"
+    _header_data
+    "${_header_data}"
+)
+file(WRITE "${CURRENT_PACKAGES_DIR}/include/FreeImage.h" "${_header_data}")
+
 vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/license-fi.txt")
