diff --git a/portfile.cmake b/portfile.cmake
index 69efaaef3..16887bb97 100644
--- a/portfile.cmake
+++ b/portfile.cmake
@@ -23,7 +23,7 @@ vcpkg_cmake_configure(
     OPTIONS
         -DJAS_ENABLE_AUTOMATIC_DEPENDENCIES=OFF
         -DJAS_ENABLE_LIBJPEG=ON
-        -DJAS_ENABLE_OPENGL=${JAS_ENABLE_OPENGL}
+        -DJAS_ENABLE_OPENGL=OFF
         -DJAS_ENABLE_DOC=OFF
         -DJAS_ENABLE_PROGRAMS=OFF
         -DJAS_ENABLE_SHARED=${JAS_ENABLE_SHARED}
diff --git a/vcpkg.json b/vcpkg.json
index 92429c150..b1e44ea77 100644
--- a/vcpkg.json
+++ b/vcpkg.json
@@ -36,10 +36,6 @@
     "opengl": {
       "description": "Enable the use of the OpenGL/GLUT Library",
       "dependencies": [
-        {
-          "name": "freeglut",
-          "platform": "!osx & !(windows & arm) & !uwp"
-        },
         {
           "name": "opengl",
           "platform": "!(windows & arm) & !uwp"
