diff --git a/no-executables-ios.patch b/no-executables-ios.patch
new file mode 100644
index 0000000..5db514a
--- /dev/null
+++ b/no-executables-ios.patch
@@ -0,0 +1,12 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 52439b3..84907e3 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -496,6 +496,7 @@ install(FILES "${CMAKE_CURRENT_BINARY_DIR}/liblzmaConfig.cmake"
+ 
+ # The command line tools needs this.
+ check_symbol_exists(getopt_long getopt.h HAVE_GETOPT_LONG)
++set(HAVE_GETOPT_LONG 0)
+ 
+ 
+ #############################################################################
diff --git a/portfile.cmake b/portfile.cmake
index e58a0e5..05b2588 100644
--- a/portfile.cmake
+++ b/portfile.cmake
@@ -7,6 +7,7 @@ vcpkg_from_github(
     PATCHES
         enable-uwp-builds.patch
         fix_config_include.patch
+        no-executables-ios.patch
         win_output_name.patch # Fix output name on Windows. Autotool build does not generate lib prefixed libraries on windows. 
 )
 
@@ -131,4 +132,4 @@ if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
 endif()
 
 file(INSTALL  ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
-file(COPY "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
\ No newline at end of file
+file(COPY "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
