diff --git a/portfile.cmake b/portfile.cmake
index c617c96..0ee6e20 100644
--- a/portfile.cmake
+++ b/portfile.cmake
@@ -3,13 +3,13 @@ vcpkg_fail_port_install(ON_ARCH "arm" ON_TARGET "uwp")
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO warmcat/libwebsockets
-    REF 736f0739ed8c75db0e29c7678b5a9194e957c347 # v4.1.6
-    SHA512 7632439f8cdc87d275a0524b87ea5cd19c3401c1b7bb7fcf673eef3e1ab36c0ec022d2dce050d07bc106af3166575882507a56ac829a4361329311484dc2c5c4
+    REF 7c3c179e0e8f9a822f3744e358dc4f62e796500b # v4.2-stable as at 2021-6-24
+    SHA512 5c90f9e90c16f75fd98829d57bb898959c9448621e06e605f3eb20853cea19953252fcb05887e46019a484dfc66100c31b408e8548644bc8ad5ab8475a1bb637
     HEAD_REF master
     PATCHES
         fix-dependency-libuv.patch
         fix-build-error.patch
-        export-include-path.patch
+        #export-include-path.patch
 )
 
 string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" LWS_WITH_STATIC)
diff --git a/vcpkg.json b/vcpkg.json
index 897f677..2915015 100644
--- a/vcpkg.json
+++ b/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "libwebsockets",
-  "version-semver": "4.1.6",
-  "port-version": 2,
+  "version-semver": "4.2.0",
+  "port-version": 0,
   "description": "Libwebsockets is a lightweight pure C library built to use minimal CPU and memory resources, and provide fast throughput in both directions as client or server.",
   "homepage": "https://github.com/warmcat/libwebsockets",
   "supports": "!(arm | uwp)",
