diff --git a/CONTROL b/CONTROL
index 71f4fe588..186e3f434 100644
--- a/CONTROL
+++ b/CONTROL
@@ -2,7 +2,7 @@ Source: qt5-base
 Version: 5.12.8-4
 Homepage: https://www.qt.io/
 Description: Qt5 Application Framework Base Module. Includes Core, GUI, Widgets, Networking, SQL, Concurrent and other essential qt components.
-Build-Depends: zlib, libjpeg-turbo, libpng, freetype, pcre2, harfbuzz, sqlite3, libpq, double-conversion, openssl, angle (!windows), egl-registry, icu (!uwp), fontconfig (!windows)
+Build-Depends: zlib, libjpeg-turbo, libpng, pcre2, sqlite3, libpq, double-conversion, openssl, angle (!windows), egl-registry, icu (!uwp), fontconfig (!windows)
 
 Feature: latest
 Description: Build latest qt version (5.14.2) instead of LTS
diff --git a/cmake/qt_download_submodule.cmake b/cmake/qt_download_submodule.cmake
index 8e03b00a8..a50e01caa 100644
--- a/cmake/qt_download_submodule.cmake
+++ b/cmake/qt_download_submodule.cmake
@@ -15,8 +15,8 @@ function(qt_download_submodule)
 
     set(FULL_VERSION "${QT_MAJOR_MINOR_VER}.${QT_PATCH_VER}")
     set(ARCHIVE_NAME "${NAME}-everywhere-src-${FULL_VERSION}.tar.xz")
-    set(URLS "http://download.qt.io/official_releases/qt/${QT_MAJOR_MINOR_VER}/${FULL_VERSION}/submodules/${ARCHIVE_NAME}"
-    "http://mirrors.ocf.berkeley.edu/qt/official_releases/qt/${QT_MAJOR_MINOR_VER}/${FULL_VERSION}/submodules/${ARCHIVE_NAME}"
+    set(URLS "http://download.qt.io/archive/qt/${QT_MAJOR_MINOR_VER}/${FULL_VERSION}/submodules/${ARCHIVE_NAME}"
+    "http://mirrors.ocf.berkeley.edu/qt/archive/qt/${QT_MAJOR_MINOR_VER}/${FULL_VERSION}/submodules/${ARCHIVE_NAME}"
     )
     vcpkg_download_distfile(ARCHIVE_FILE
         URLS ${URLS}
diff --git a/portfile.cmake b/portfile.cmake
index 984ff123b..a2d1f85d4 100644
--- a/portfile.cmake
+++ b/portfile.cmake
@@ -95,11 +95,11 @@ list(APPEND CORE_OPTIONS
     -system-zlib
     -system-libjpeg
     -system-libpng
-    -system-freetype # static builds require to also link its dependent bzip!
+    -no-freetype # static builds require to also link its dependent bzip!
     -system-pcre
     -system-doubleconversion
     -system-sqlite
-    -system-harfbuzz
+    -no-harfbuzz
     -icu
     -no-angle)      # Qt does not need to build angle. VCPKG will build angle!
 
