--- jsep.h	2019-10-18 09:10:15.000000000 +0200
+++ jsep_patched.h	2019-10-18 10:09:36.000000000 +0200
@@ -221,8 +221,13 @@
   // is deprecated; in order to let clients remove the old version, it has a
   // default implementation. If both versions are unimplemented, the
   // result will be a runtime error (stack overflow). This is intentional.
-  virtual void OnFailure(RTCError error);
-  virtual void OnFailure(const std::string& error);
+  void OnFailure(RTCError error) {
+    OnFailure(error.message());
+  }
+
+  void OnFailure(const std::string& error) {
+    OnFailure(RTCError(RTCErrorType::INTERNAL_ERROR, std::string(error)));
+  }
 
  protected:
   ~CreateSessionDescriptionObserver() override = default;
@@ -233,9 +238,14 @@
  public:
   virtual void OnSuccess() = 0;
   // See description in CreateSessionDescriptionObserver for OnFailure.
-  virtual void OnFailure(RTCError error);
-
-  virtual void OnFailure(const std::string& error);
+  void OnFailure(RTCError error) {
+    std::string message(error.message());
+    OnFailure(message);
+  }
+
+  void OnFailure(const std::string& error) {
+    OnFailure(RTCError(RTCErrorType::INTERNAL_ERROR, std::string(error)));
+  }
 
  protected:
   ~SetSessionDescriptionObserver() override = default;
