diff --git a/core/fpdfapi/page/cpdf_annotcontext.cpp b/core/fpdfapi/page/cpdf_annotcontext.cpp
index 205a77c1e..0a13c5f55 100644
--- a/core/fpdfapi/page/cpdf_annotcontext.cpp
+++ b/core/fpdfapi/page/cpdf_annotcontext.cpp
@@ -4,8 +4,6 @@

 // Original code copyright 2014 Foxit Software Inc. http://www.foxitsoftware.com

-#include "core/fpdfapi/page/cpdf_annotcontext.h"
-
 #include <utility>

 #include "core/fpdfapi/page/cpdf_form.h"
@@ -13,6 +11,7 @@
 #include "core/fpdfapi/parser/cpdf_dictionary.h"
 #include "core/fpdfapi/parser/cpdf_stream.h"
 #include "third_party/base/check.h"
+#include "core/fpdfapi/page/cpdf_annotcontext.h"

 CPDF_AnnotContext::CPDF_AnnotContext(RetainPtr<CPDF_Dictionary> pAnnotDict,
                                      IPDF_Page* pPage)

