diff --git a/Source/FreeImage.h b/Source/FreeImage.h
index ad2574d..dd541b4 100644
--- a/Source/FreeImage.h
+++ b/Source/FreeImage.h
@@ -155,7 +155,15 @@ FI_STRUCT (FIMULTIBITMAP) { void *data; };
 #ifndef _MSC_VER
 // define portable types for 32-bit / 64-bit OS
 #include <inttypes.h>
+#ifdef __APPLE__
+#if OBJC_BOOL_IS_BOOL // from objc.h
+typedef bool BOOL;
+#else
+typedef signed char BOOL;
+#endif
+#else
 typedef int32_t BOOL;
+#endif
 typedef uint8_t BYTE;
 typedef uint16_t WORD;
 typedef uint32_t DWORD;
